blog/2021-07-17-introducing-metriql.md

```diff
@@ -0,0 +1,94 @@
+
+```
+sales - LAG(sales) OVER (order by month)
+```
+
+Your business stakeholder wants to use different data buckets such as quarter and year and you realize that you need to create different data models for each of them. It’s OK, you will just copy and paste the existing month and change the date bucket accordingly. After some time, the business stakeholder wants to use different dimensions such as user country, segment, etc. and you realize that you should use Tableau calculations to be able to make it dynamic. Since you already know about the LAG function, you start looking at the Tableau-way but [it’s](https://community.tableau.com/s/question/0D54T00000C6lOy/lag-function-logic) [completely](https://community.tableau.com/s/question/0D54T00000C6afV/how-to-do-lag-in-tableau) [different](https://community.tableau.com/s/question/0D54T00000C6XqLSAV/calculating-lag-days) than SQL approach and now you need to maintain that metric inside Tableau while you’re using SQL for the rest of the data model.
+
+While you can cover these use-cases in most of the well-known BI tools, there are two issues:
+
+1. You need to learn yet another expression language that’s never flexible enough compared to SQL. (It must be compiled to SQL anyway)
+2. Since you usually need to use a GUI to define the metrics, it’s not easy to automate, test, or share them with other data tools. (I’m not even talking about continuous integration and deployment)
+
+Moreover, the problem is actually not just about the metric definitions. It's also about another buzzword everyone is talking about, self-serve analytics. If you aim to be one of those data-driven companies, you should enable all the people in your organization to analyze the data without any assistance from the data teams. To achieve that,  depending on the organizational structure, you can either:
+
+1. Teach everyone SQL and have a company-wide documentation for your data.
+2. Hire a data team, collect all the company data into a data warehouse, model the data for business requirements and expose the data to the business stakeholders using their preferred tools.
+
+It's not realistic to expect every company to have the luxury to teach SQL to all their stakeholders. Many departments such as product, marketing, and finance have different sets of priorities and responsibilities and don't have [time](https://twitter.com/sethrosen/status/1363327369726074880) to invest in learning SQL. If you go with the latter approach, you usually need to use an ELT provider such as Fivetran, Stitch, etc. and adopt a data transformation tool such as dbt or Airflow for your company or stick with the data transformation features of your BI tool. You usually have the following two options with BI:
+
+##### Go through the sales process and buy an enterprise product such as Tableau, PowerBI, etc.:
+
+* Advantages
+    * Powerful visualization
+    * Ability to analyze the data much faster if your data source is slow
```

Comment:
Ability to analyze the data much faster if your data source is slow - by extracting the data olarak mi diyorsun?
Bu enterprise BI in altına oluyor mu looker da girer diyoruz çünkü asagida enterprise Bi a

Reply:
Looker yapmiyo bunu haklisin, oraya bi not ekleyecegim. Extracting data kismini da eklesek iyi olur anlam daha gucleniyo.

Reply:
```suggestion
    * Ability to analyze the data much faster by extracting the data if your data source is slow
```